include/experimental/kernel/col_major_shuf/col_major_shuf_xe.hpp

```diff
@@ -0,0 +1,152 @@
+/*******************************************************************************
+ * Copyright (c) 2023-2024 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+/// @file
+/// C++ API
+
+#pragma once
+
+#include <experimental/kernel/col_major_shuf/api.hpp>
+#include <experimental/kernel/col_major_shuf/common.hpp>
+#include <experimental/kernel/col_major_shuf/config.hpp>
+
+namespace gpu::xetla::kernel {
+template <
+    typename dtype_in_,
+    typename dtype_out_,
+    typename dtype_gidx_,
+    mem_layout mem_layout_in_,
+    typename col_major_shuf_attr_,
+    gpu_arch arch_>
+struct col_major_shuf_t<
+    dtype_in_,
+    dtype_out_,
+    dtype_gidx_,
+    mem_layout_in_,
+    col_major_shuf_attr_,
+    arch_,
+    std::enable_if_t<arch_ == gpu_arch::XeHpg || arch_ == gpu_arch::XeLpg>> {
```

Comment:
why gpu_arch is limited?

Reply:
will you implement a variant using 2d block load/store? If yes, change this to ```!arch_has_2d_load_store```

Reply:
1. because we only validated on MTL & ARC, but it should work fine on PVC;
2. in shuffle kernel we use gather instructions to load data and seems that hardware dosen't provide 2d-gather-load capability; for store, we use tile_store to store data, and the specialization determined by arch_ template param, if arch_==XeHpc, kernel will use 2d_store instructions.

Reply:
then remove the limit

Reply:
done in [d9ce5b1](https://github.com/intel/neural-speed/pull/235/commits/d9ce5b187d0bec1cafc1d766e6d209870bc1e9b8)